sklearn/linear_model/_stochastic_gradient.py

```diff
@@ -280,10 +280,10 @@ def _make_validation_split(self, y):
         Returns
         -------
         validation_mask : ndarray of shape (n_samples, )
-            Equal to 1 on the validation set, 0 on the training set.
+            Equal to True on the validation set, False on the training set.
         """
         n_samples = y.shape[0]
-        validation_mask = np.zeros(n_samples, dtype=np.uint8)
+        validation_mask = np.zeros(n_samples, dtype=bool)
```

Comment:
Let's use `np.bool_` here to be careful with the Cython code:

```suggestion
        validation_mask = np.zeros(n_samples, dtype=np.bool_)
```

**Details**: I think this was done this way to match the signature of `_plain_sgd`:

https://github.com/scikit-learn/scikit-learn/blob/920ab2508fe634ad32df68bb0ebd4f4512fbfb53/sklearn/linear_model/_sgd_fast.pyx#L374

but using numpy's `bool` is safe because it is typedefed to a [unsigned char](https://github.com/numpy/numpy/blob/a4a757d3b971a6b25608a91e9985c4ff0b10dc82/numpy/__init__.pxd#L253).